cmd/algoh/main.go

```diff
@@ -126,11 +127,32 @@ func main() {
 
 		err = cmd.Start()
 		if err != nil {
-			reportErrorf("Error starting algod: %v", err)
+			reportErrorf("error starting algod: %v", err)
+		}
+		err = cmd.Wait()
+		if err != nil {
+			reportErrorf("error waiting for algod: %v", err)
 		}
-		cmd.Wait()
 		close(done)
 
+		// capture logs if algod terminated prior to blockWatcher starting
+		if !blockWatcherInitialized && algohConfig.UploadOnError {
+			if errorOutput.output != "" {
```

Comment:
This looks like a duplicate of the code below - please consolidate.  Also - this block should not all be gated by 'UploadOnError' - that should only affect whether we call sendLogs(), like below.

Reply:
OK,Refactored into a separate method collectErrorLogs().  Removed UploadOnError condition.